temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -182,12 +194,42 @@ private PollerOptions getPollerOptions(SingleWorkerOptions options) {
       pollerOptions =
           PollerOptions.newBuilder(pollerOptions)
               .setPollThreadNamePrefix(
-                  WorkerThreadsNameHelper.getWorkflowPollerThreadPrefix(namespace, taskQueue))
+                  WorkerThreadsNameHelper.getWorkflowPollerThreadPrefix(namespace, taskQueueName))
               .build();
     }
     return pollerOptions;
   }
 
+  @Nullable
+  public WorkflowTaskDispatchHandle reserveWorkflowExecutor() {
+    // to avoid pollTaskExecutor to become null inside the lambda, we are caching it here
+    final PollTaskExecutor<WorkflowTask> executor = pollTaskExecutor;
+    return executor != null && !isSuspended() && wftExecutorSlotsSemaphore.tryAcquire()
+        ? new WorkflowTaskDispatchHandle(
+            workflowTask -> {
+              String queueName =
+                  workflowTask.getResponse().getWorkflowExecutionTaskQueue().getName();
+              TaskQueueKind queueKind =
+                  workflowTask.getResponse().getWorkflowExecutionTaskQueue().getKind();
+              Preconditions.checkArgument(
+                  this.taskQueueName.equals(queueName)
+                      || TaskQueueKind.TASK_QUEUE_KIND_STICKY.equals(queueKind)
```

Comment:
Right now there's no way you'll get back an eager task on a sticky queue but I agree it doesn't hurt to accept it.